tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -239,13 +239,22 @@ private void OnEndRequest(object sender, EventArgs eventArgs)
 
                         if (!rootSpan.HasHttpStatusCode())
                         {
-                            var status = app.Context.Response.StatusCode;
-                            rootSpan.SetHttpStatusCode(status, isServer: true, Tracer.Instance.Settings);
+                            var response = app.Context.Response;
+                            var status = response.StatusCode;
+                            var isServer = true;
+                            if (status == 500 && !response.IsClientConnected && app.Context.Error is null)
+                            {
+                                // rare case when client disconnects - IIS returns a 400 but reports a 500
+                                status = 400;
+                                isServer = false;
```

Comment:
Why are we setting `isServer = false`? So the span still gets tagged with an error?

Reply:
So my thinking was: this error is a _client_ error; the client disconnected. The `isServer` variable determines whether we use the `HttpServerErrorStatusCodes` or `HttpClientErrorStatusCodes` settings to decide whether the error should actually be marked as an error or not.

Turns out, that's completely wrong 😅 `HttpServerErrorStatusCodes` and `HttpClientErrorStatusCodes` don't have anything to do with whether it's a server or client _error_, just whether it's the server or client that's creating the span. 

Fixed, thanks!